ERCS/erc-7730.md

```diff
@@ -0,0 +1,1151 @@
+
+Hardware wallets tend to have more limited capabilities that will impact what they can display securely, especially since the intention of the specification is to create a strong security binding between the spec and the data being reviewed.
+
+This consideration is driving a few choices done for ERC-7730:
+* Complex UI constructs like layouts, grouping and re-ordering of fields have been left to a wallet specific section, yet unspecified. After a time, we may see patterns emerge between wallets in terms of minimal features.
+* Representation of fields has been created to allow "flattening" the list of fields, when handling complex message structures. This flattened representation is expected to work better with Hardware wallets in particular, and is recommended at first.
+* Some formatters that require recursive constructs, like `calldata` are expected to work with restrictions at first, especially on Hardware wallets.  
+
+### Why is the ABI still necessary? 
+
+The full signature of a function call does not contain information about the types of complex parameters that might be used in the function call, only their names, while the ABI does carry this information. This is why the ABI is the reference and the function signature MUST match it. The full form of the function call is allowed in order to simplify writing and reading the ERC-7730 file. 
+
+### Deployment model
+ 
+Making the ERC-7730 available for wallets is a key factor of adoption. We have a few options:
+- In each dApps GitHub repository or website: good for autonomy but can be problematic for discoverability by wallets of new ERC-7730 files
+- Foundation operated repository, like ethereum chainID list: good alternative between decentralization and discoverability.
+- Ledger repository: as a short term solution, Ledger is providing a central repository (See Ledger GitHub repository)
+
+### Extensibility to other Structured Data formats
+
+In the future, this specification could be extended to structured data like Meta Transaction in [EIP-2771](./eip-2771.md) or User Operations in [EIP-4337](./eip-4337.md).
+
+## Test Cases
+
+More examples can be found in the asset folder.
+
+## Security Considerations
+
+The main security concern introduced by ERC-7730 is to avoid attacks that would use the ERC-7730 formatting mechanism to trick users into signing something wrong.
+
+Two main attack vectors that needs to be mitigated:
+* Avoid using a well-formed, trusted ERC-7730 file on data that should not be formatted using it
+* Avoid badly formed, incorrect ERC-7730 files that would hide critical parameters or create misinterpretations through bed formatting
+
+### Binding context
+
+The binding `context` is the way we mitigate the first attack, by specifying in the ERC-7730 file itself, **what kind** of structured data should be formatted by said file.
+
+dApps developers MUST make sure that constraints in the binding context are restrictive enough to apply only to relevant data.
```

Comment:
You should keep requirements (the all caps keywords) to the specification section.